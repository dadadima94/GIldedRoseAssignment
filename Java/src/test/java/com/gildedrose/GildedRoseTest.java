package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

class GildedRoseTest {

    @Test
    void updateQualityMethod() {

        /**
         * testing all the possible combinations that could generate bugs: verifyAllCombination method of Approvals tests
         *
         * Elements found with Code Coverage: https://en.wikipedia.org/wiki/Code_coverage
         *  - Strings: {"Standard Item", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros"},
         *  - Integer:  {-1, 0, 10},
         *  - Integer:  {0, 1, 49, 50}
         *
         *  Elements found with Mutation Testing: https://en.wikipedia.org/wiki/Code_coverage
         *
         *
         */
        CombinationApprovals.verifyAllCombinations(this::newItemAndUpdateQuality,
                new String[]{"Standard Item", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros"},
                new Integer[]{-1, 0, 10},
                new Integer[]{0, 1, 49, 50});
    }

    private String newItemAndUpdateQuality(String itemName, int itemSellIn, int itemQuality) {
        Item[] items = new Item[] { new Item(itemName, itemSellIn, itemQuality) };
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        return app.items[0].toString();
    }

}
