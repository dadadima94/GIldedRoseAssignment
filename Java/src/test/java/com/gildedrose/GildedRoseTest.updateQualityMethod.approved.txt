[Standard Item, -1, 0] => Standard Item, -2, 0 
[Standard Item, -1, 1] => Standard Item, -2, 0 
[Standard Item, -1, 49] => Standard Item, -2, 47 
[Standard Item, -1, 50] => Standard Item, -2, 48 
[Standard Item, 0, 0] => Standard Item, -1, 0 
[Standard Item, 0, 1] => Standard Item, -1, 0 
[Standard Item, 0, 49] => Standard Item, -1, 47 
[Standard Item, 0, 50] => Standard Item, -1, 48 
[Standard Item, 11, 0] => Standard Item, 10, 0 
[Standard Item, 11, 1] => Standard Item, 10, 0 
[Standard Item, 11, 49] => Standard Item, 10, 48 
[Standard Item, 11, 50] => Standard Item, 10, 49 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 1] => Aged Brie, -2, 3 
[Aged Brie, -1, 49] => Aged Brie, -2, 50 
[Aged Brie, -1, 50] => Aged Brie, -2, 50 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 1] => Aged Brie, -1, 3 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, 11, 0] => Aged Brie, 10, 1 
[Aged Brie, 11, 1] => Aged Brie, 10, 2 
[Aged Brie, 11, 49] => Aged Brie, 10, 50 
[Aged Brie, 11, 50] => Aged Brie, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => Backstage passes to a TAFKAL80ETC concert, 10, 1 
[Backstage passes to a TAFKAL80ETC concert, 11, 1] => Backstage passes to a TAFKAL80ETC concert, 10, 2 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, 1] => Sulfuras, Hand of Ragnaros, -1, 1 
[Sulfuras, Hand of Ragnaros, -1, 49] => Sulfuras, Hand of Ragnaros, -1, 49 
[Sulfuras, Hand of Ragnaros, -1, 50] => Sulfuras, Hand of Ragnaros, -1, 50 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 1] => Sulfuras, Hand of Ragnaros, 0, 1 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 49 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50 
[Sulfuras, Hand of Ragnaros, 11, 0] => Sulfuras, Hand of Ragnaros, 11, 0 
[Sulfuras, Hand of Ragnaros, 11, 1] => Sulfuras, Hand of Ragnaros, 11, 1 
[Sulfuras, Hand of Ragnaros, 11, 49] => Sulfuras, Hand of Ragnaros, 11, 49 
[Sulfuras, Hand of Ragnaros, 11, 50] => Sulfuras, Hand of Ragnaros, 11, 50 
